rdt/transformers/numerical.py

```diff
@@ -104,7 +104,7 @@ def _raise_out_of_bounds_error(self, value, name, bound_type, min_bound, max_bou
 
     def _validate_values_within_bounds(self, data):
         if not self.computer_representation.startswith('Float'):
-            fractions = data[~data.isna() & data % 1 != 0]
+            fractions = data[~data.isna() & (data != (data // 1))]
```

Comment:
Can we add unit & integration tests to this and the other function? To verify that we support the numpy data types and pyarrow data types?